tools/scripts/autorun_generateSoundMetadata.rb

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env ruby
+
+# FOR USE WHEN WORKIN WITH MULTIPLE SOUND FILES IN A FOLDER
+# FOR GENERATING METDATA FOR A SINGLE FILE THEN USE: generateSoundMetadata.rb
+# Use this script for converting multiple audio formats into .mp3 (required)
+# Then generateSoundMetadata.rb will be run to:
+# 1. create metadata (JSON) for each file (required)
+# 2. use file_path to get the directory name like animals or notifications
+# to add category_animals or category_notifications to the aliases (required)
+
+# command to run file: bundle exec ./autorun_generateSoundMetadata.rb [folder_path] [csv_location]
+# example for folder_name = Alert
+# example for csv_location = ./sound_metadata.csv
+# example command = bundle exec ./parse_new_sound_aliases_csv.rb ~/Downloads/Animals ./sound_metadata.csv
+require 'csv'
+require 'fileutils'
+
+class NewSoundAliasesMetadataBuilder
+  def initialize(csv)
+    @csv = CSV.parse(File.read(csv), headers: true)
+  end
+
+  def fix_corrupted_filename(name, type_of_file)
+    name.downcase! if name =~ /[A-Z]/
+    # some filenames had spaces instead of underscores
+    name.tr!(" ", "_") if name.include?(" ")
+    # some filenames have more than one underscore but we want to match
+    #  the style of existing sound filenames with one underscore
+    name.gsub!(/_{2,}/, "_") if name =~ /_{2,}/
+    name.gsub!(/^\_+/, '') if name =~ /^\_+/
+
+    if name.chomp([-4..-1].to_s) =~ /\_+$/
+      name.gsub!(/\_+$/, '')
+      name.chomp!([-4..-1].to_s)
+      name << ".mp3" if type_of_file == "csv"
+    end
+
+    return name
```

Comment:
Got it! I removed redundancies by removing all conditionals for that regex in the `fix_corrupted_filename` function since they are mutating the `name` parameter anyway and I'll rename the function as `fix_corrupted_filename!`.